frontend/src/common/queries/wheresMyGene.ts

```diff
@@ -769,37 +762,62 @@ export interface MarkerGenesByCellType {
 
 export interface MarkerGeneResponse {
   marker_genes: {
-    [geneID: string]: {
+    // key is gene id from backend, but we convert to gene name
+    [key: string]: {
       effect_size: number;
       p_value: number;
     };
   };
   snapshot_id: string;
 }
 
-export function useMarkerGenes(genesByGeneID: {
-  [id: string]: Gene;
-}): UseMutationResult<MarkerGeneResponse, unknown, FetchMarkerGeneParams> {
-  const queryClient = useQueryClient();
-  const dispatch = useContext(DispatchContext);
-  const prevMarkerGenes = queryClient.getQueryData<MarkerGenesByCellType>([
-    USE_MARKER_GENES,
-  ]);
-  return useMutation(fetchMarkerGenes, {
-    onSuccess: (data, { cellTypeID }) => {
-      if (!dispatch || !data) return;
-      const newMarkerGenes = prevMarkerGenes || {};
-      newMarkerGenes[cellTypeID] = data.marker_genes;
-      queryClient.setQueryData([USE_MARKER_GENES], newMarkerGenes);
-      const geneNames = Object.keys(data.marker_genes).map((geneID) => {
-        const gene = genesByGeneID[geneID];
-        if (!gene) {
-          console.error("gene not found", geneID);
-        }
-        return gene.name;
+export function useMarkerGenes({
+  cellTypeID,
+  organismID,
+  tissueID,
+  test,
+}: FetchMarkerGeneParams): UseQueryResult<MarkerGeneResponse> {
+  const { data } = usePrimaryFilterDimensions();
+  const genesByID = useMemo((): { [name: string]: OntologyTerm } => {
+    let result: { [name: string]: OntologyTerm } = {};
+
+    if (!data) return result;
+
+    const { genes } = data;
+
+    result = generateTermsByKey(genes, "id");
+
+    return result;
+  }, [data]);
+
+  return useQuery(
+    [USE_MARKER_GENES, cellTypeID, test],
+    // Getting some weird issues with fetching some times (right side bar is not populated),
+    //  this may be related to receiving NaN's from the backend. Going to ignore for now
```

Comment:
Is this issue still relevant / reproducible?

When right sidebar isn't being populated, is it consistent for the same cell type query? If it's not consistent, then it's likely not to do with the backend.